lms/static/js/spec/student_account/account_settings_factory_spec.js

```diff
@@ -148,4 +148,158 @@ define(['backbone',
+            });
+
+
+            it('shows loading error when UserPreferencesModel fails to load for enterprise learners', function() {
+                var accountSettingsView, request;
+                requests = AjaxHelpers.requests(this);
+
+                accountSettingsView = createEnterpriseLearnerAccountSettingsPage();
+
+                Helpers.expectLoadingErrorIsVisible(accountSettingsView, false);
+
+                request = requests[0];
+                expect(request.method).toBe('GET');
+                expect(request.url).toBe(Helpers.USER_ACCOUNTS_API_URL);
+
+                AjaxHelpers.respondWithJson(requests, Helpers.createAccountSettingsData());
+                Helpers.expectLoadingErrorIsVisible(accountSettingsView, false);
+
+                request = requests[1];
+                expect(request.method).toBe('GET');
+                expect(request.url).toBe('/user_api/v1/preferences/time_zones/?country_code=1');
+                AjaxHelpers.respondWithJson(requests, Helpers.TIME_ZONE_RESPONSE);
+
+                request = requests[2];
+                expect(request.method).toBe('GET');
+                expect(request.url).toBe(Helpers.USER_PREFERENCES_API_URL);
+
+                AjaxHelpers.respondWithError(requests, 500);
+                Helpers.expectLoadingErrorIsVisible(accountSettingsView, true);
+            });
+
+            it('renders fields after the models are successfully fetched for enterprise learners', function() {
+                var accountSettingsView;
+                requests = AjaxHelpers.requests(this);
+
+                accountSettingsView = createEnterpriseLearnerAccountSettingsPage();
+
+                Helpers.expectLoadingErrorIsVisible(accountSettingsView, false);
+
+                AjaxHelpers.respondWithJson(requests, Helpers.createAccountSettingsData());
+                AjaxHelpers.respondWithJson(requests, Helpers.TIME_ZONE_RESPONSE);
+                AjaxHelpers.respondWithJson(requests, Helpers.createUserPreferencesData());
+
+                accountSettingsView.render();
+
+                Helpers.expectLoadingErrorIsVisible(accountSettingsView, false);
+                Helpers.expectSettingsSectionsAndFieldsToBeRenderedWithMessage(accountSettingsView);
+            });
+
+            it('expects all fields to behave correctly for enterprise learners', function() {
+                var accountSettingsView, i, view, sectionsData, textFields, dropdownFields;
+                requests = AjaxHelpers.requests(this);
+
+                accountSettingsView = createEnterpriseLearnerAccountSettingsPage();
+
+                AjaxHelpers.respondWithJson(requests, Helpers.createAccountSettingsData());
+                AjaxHelpers.respondWithJson(requests, Helpers.TIME_ZONE_RESPONSE);
+                AjaxHelpers.respondWithJson(requests, Helpers.createUserPreferencesData());
+                AjaxHelpers.respondWithJson(requests, {});  // Page viewed analytics event
```

Comment:
Why analytics event required in that test?

Reply:
These checks were implemented in existing code so I also move them in new tests.